minject/registry.py

```diff
@@ -86,9 +102,35 @@ def config(self) -> RegistryConfigWrapper:
     def resolve(self, key: "RegistryKey[T]") -> T:
         return self[key]
 
+    async def _aresolve(self, key: "RegistryKey[T]") -> T:
```

Comment:
It would be helpful to have doc comments for these helper functions — in the existing code it is already not clear to me why we need both `get` and `resolve`, and so it's also not clear to me why we need both `aget` and `_aresolve`.

Reply:
I'm also not sure if both are needed, and I bet you are right that `resolve/_aresolve` could be removed.

I'd prefer to not clean this up as part of this PR however in order to keep the PR focused and to increase the probability of hitting deadlines surrounding this async support PR.

I've created TODO ISSUE to track looking into simplifying the `resove`/`_aresolve` indirection.

Reply:
This seems correct. Agree it's a bit crazy. Making the mutual recursion a bit more straightforward would probably be worth doing.

I'll make an issue to document this refactor idea: